nautobot/docs/development/core/model-checklist.md

```diff
@@ -0,0 +1,167 @@
+# Model Development Checklist
+
+A general best-practices checklist to follow when adding a new data model in Nautobot or adding new fields to an existing model.
+
+<!-- markdownlint-disable no-inline-html -->
+<style>
+article ul li:before {
+    content: '□';
+    margin:0 5px 0 -15px;
+}
+article ul li {
+    list-style-type: none;
+}
+</style>
+<!-- markdownlint-enable no-inline-html -->
+
+## Bootstrapping a new Model
+
+### Data Model
+
+- Implement model in `nautobot.<app>.models` module
+    - Use appropriate [base class](best-practices.md#base-classes) and mixin(s)
+    - Use appropriate `@extras_features` decorator values
+    - Define appropriate uniqueness constraint(s)
+    - Define appropriate `__str__()` logic
+    - _optional_ Define appropriate additional [`clean()`](best-practices.md#model-validation) logic
+        - Be sure to always call `super().clean()` as well!
+    - _optional_ Define appropriate `verbose_name`/`verbose_name_plural` if defaults aren't optimal
+- Generate database schema migration(s) with `invoke makemigrations <app> -n <migration_name>`
+- _optional_ Add [data migration(s)](https://docs.djangoproject.com/en/stable/topics/migrations/) to populate default records, migrate data from existing models, etc.
+    - Remember: data migrations must not share a file with schema migrations or vice versa!
+    - Specify appropriate migration dependencies (for example, against another app that you're using models from)
```

Comment:
Can we add some context for this common mistake?

> Any model retrieved in a data migration with `apps.get_model("app_label.model_name")` should have a dependency that ensures the model is in the desired state before your migration runs. This may only require a dependency on the migration where that model was created or, if a specific field is required on the referenced model, the migration where that field was introduced.